azurelinuxagent/distro/suse/rdma.py

```diff
@@ -34,11 +34,14 @@ def install_driver(self):
             error_msg += 'no driver will be installed.'
             logger.error(error_msg)
             return
-        zypper_install = 'zypper -n in'
-        zypper_search = 'zypper se -s'
+        zypper_install = 'zypper -n in %s'
+        zypper_remove = 'zypper -n rm %s'
+        zypper_search = 'zypper se -s %s'
```

Comment:
Well it is not necessarily about finding what is installed, it is also about finding what is available. The "zypper se -s" call gives us both. We have to cover 3 conditions:

1.) No driver installed => install the proper driver version from the available packages, thus I have to find all the avialbale packages
2.) Driver installed and matching version => do nothing (this case could be covered by the rpm call from the 2.0 branch
3.) Driver installed but the FW version does not match => uninstall the driver and find the proper version from all the available packages

All cases are covered by the information zypper returns. Only 1 case is covered by the information the rpm call returns.